taipy/core/_entity/submittable.py

```diff
@@ -86,7 +88,10 @@ def is_ready_to_run(self) -> bool:
         Returns:
             True if the given entity is ready to be run. False otherwise.
         """
-        return all(dn.is_ready_for_reading for dn in self.get_inputs())
+        if self._submittable_id not in _ReadyToRunCache._submittable_id_datanodes:
+            for dn in self.get_inputs():
+                _compute_if_dn_is_ready_for_reading(dn)
+        return _ReadyToRunCache._check_submittable_is_ready_to_submit(self._submittable_id)
```

Comment:
2 reasons why I decided to put it here:
- A way to trigger tracking the submittable status of the scenario
- To make sure, in case the Gui team check but the scenario is not submittable yet, it's still loaded into the cache so that when it's available for submission, it can notify the user/gui about that

Reply:
we will rollback due to the unneeded initiation of the cache